include/core/jmanagedbackends.h

```diff
@@ -0,0 +1,399 @@
+
+	/// short processing period to match new object to a storage tier
+	/** \retval FALSE on error
+	 * \sa j_backend_stack_get_tiers */
+	/**
+	 * Signals a new object creation to the policy, sets instal storage tier.
+	 *
+	 * \param[inout] policy_data pointer usage depends on policy
+	 * \param[in] namespace of the object which will be created
+	 * \param[in] path of the object which will be created
+	 * \param[in] obj_id \ref process_access
+	 * \param[out] storage_tier ID of storage tier to create object on
+	 **/
+	gboolean (*process_create)(gpointer policy_data, const gchar* namespace, const gchar* path, guint obj_id, guint* storage_tier);
+
+	/**
+	 * handle messages send to policy.
+	 *
+	 * \todo add reply functionality
+	 *
+	 * \param[inout] policy_data pointer usage depends on policy
+	 * \param[in] type message type encoded as 0-terminated string
+	 * \param[inout] data optional message data
+	 * \param[in] size of data field
+	 **/
+	gboolean (*process_message)(gpointer policy_data, const gchar* type, gpointer data, guint length);
+
+	/**
+	 * Function use to pass a dedicated thread for migrations
+	 * and calculations to policy.
+	 **/
+	gboolean (*process)(gpointer policy_data);
+
+	gpointer data; ///< reference to data allocated in init
+} JObjectBackendPolicy;
+
+JObjectBackendPolicy* backend_policy_info(void);
+
+/**
+ * \defgroup JObjectBackendAccessDetails
+ *
+ * Structs used to pass additional data on a
+ * JObjectBackendPolicy.process_access  call
+ *
+ * @{
+ *
+ **/
+
+/**
+ * Additional data for a read and write access
+ **/
+struct JObjectBackendRWAccess
+{
+	guint64 offset; ///< access offset
+	guint64 length; ///< length of data to be written/read
+};
+
+/**
+ * @}
+ **/
+
+/**
+ * \param[in] config with information as about backend performances and policy to use
+ * \param[in] object_backends a list instantiated object backends to use, in order as configured in config
+ * \param[out] instance_ptr address of initialized JManagedBackends
+ **/
+gboolean j_backend_managed_init(JConfiguration* config, JList* object_backends, JManagedBackends** instance_ptr);
+
+gboolean j_backend_managed_fini(JManagedBackends* this);
+
+/**
+ * Hands current thread to policy as processing resource
+ * \sa JObjectBackendPolicy
+ *
+ * \param[in] this JManagedBackends instance
+ * \param[out] keep_running signals if the policy processing function should be called again after return
+ *             \todo more useful if different calling pattern get defined
+ **/
+gboolean j_backend_managed_policy_process(JManagedBackends* this, gboolean* keep_running);
+
+/**
+ * Fetches backend where a given object is stored. Also blocks migration for that object.
+ * Use j_backend_managed_object_open() to allow migration again. If the object not already exists
+ * use j_backend_managed_object_create() to create it
```

Comment:
open does also create if non-existent as implemented in *.c. Maybe this should be mentioned here. Otherwise someone could think open would fail on non-existant objects (and in worst-case use it as existence check).